### PR TITLE
✨Maintenance: enable mypy in storage

### PR DESCRIPTION
**Pull request by sanderegg**
## Related issue/s
relates to https://github.com/ITISFoundation/osparc-simcore/issues/4071

## How to test

```cmd
cd services/storage
make install-dev
make mypy
```

### CODE REVIEW

File: services/storage/src/simcore_service_storage/handlers_files.py [hunk 1]
```diff
@@ -314,8 +342,10 @@ async def is_completed_upload_file(request: web.Request):
     )
 
 
-@routes.delete(f"/{api_vtag}/locations/{{location_id}}/files/{{file_id}}", name="delete_file")  # type: ignore
-async def delete_file(request: web.Request):
+@routes.delete(
+    f"/{api_vtag}/locations/{{location_id}}/files/{{file_id}}", name="delete_file"
+)
+async def delete_file(request: web.Request) -> NoReturn:
```
Comment (pcrespov) [inline services/storage/src/simcore_service_storage/handlers_files.py:348]: ``NoReturn`` ... interesting! 🤔 
Nice, so it differentiates between returning a None or not returning anything at all (i.e. always raises) 

https://peps.python.org/pep-0484/#the-noreturn-type

File: .github/workflows/ci-testing-deploy.yml [hunk 2]
```diff
@@ -384,6 +384,8 @@ jobs:
         run: ./ci/helpers/show_system_versions.bash
       - name: install
         run: ./ci/github/unit-testing/storage.bash install
+      - name: typecheck
+        run: ./ci/github/unit-testing/storage.bash typecheck
       - name: test
         run: ./ci/github/unit-testing/storage.bash test
```
Comment (pcrespov) [inline .github/workflows/ci-testing-deploy.yml:390]: do not forget to add that ``test`` runs even if ``typecheck`` fails. See the others ``typecheck``)

**Review by pcrespov — APPROVED** [review]

File: .github/workflows/ci-testing-deploy.yml (hunk 2 above)
Comment (sanderegg) [reply .github/workflows/ci-testing-deploy.yml:390]: hmm I copy pasted that. so I will then fix it everywhere

File: services/storage/src/simcore_service_storage/datcore_adapter/datcore_adapter.py [hunk 3]
```diff
@@ -64,7 +64,15 @@ async def _request(
...
         ) as response:
-            return await response.json()
+            response_data = await response.json()
+            assert isinstance(
+                response_data,
+                (
+                    dict,
+                    list,
+                ),
+            )  # nosec
```
Comment (GitHK) [inline services/storage/src/simcore_service_storage/datcore_adapter/datcore_adapter.py:74]: can we get away with a one liner here?
```suggestion
            assert isinstance(response_data, (dict, list))  # nosec
```

File: services/storage/src/simcore_service_storage/datcore_adapter/datcore_adapter.py [hunk 4]
```diff
@@ -224,7 +233,7 @@ async def get_file_download_presigned_link(
         dict[str, Any],
         await _request(app, api_key, api_secret, "GET", f"/files/{file_id}"),
     )
-    return parse_obj_as(AnyUrl, file_download_data["link"])
+    return cast(AnyUrl, parse_obj_as(AnyUrl, file_download_data["link"]))  # mypy
```
Comment (GitHK) [inline services/storage/src/simcore_service_storage/datcore_adapter/datcore_adapter.py:236]: what is the `#mypy` comment for? Does mypy interpret it somehow?

Comment (GitHK) [reply services/storage/src/simcore_service_storage/datcore_adapter/datcore_adapter.py:236]: If it does not, I don't think it's that useful. I would already assume the casting is due to mypy.

I would just remove them all over the place where you added them.

**Review by GitHK — APPROVED** [review]

File: services/storage/src/simcore_service_storage/datcore_adapter/datcore_adapter.py (hunk 4 above)
Comment (sanderegg) [reply services/storage/src/simcore_service_storage/datcore_adapter/datcore_adapter.py:236]: well to my understanding these casting are **only** due to mypy. they are semantically correct.
therefore the comment to explain why it is done this way.

**Pull request merged**